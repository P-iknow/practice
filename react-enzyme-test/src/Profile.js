import React from 'react';

function Profie({ userName, name }) {
	return (
		<div>
			<b>{userName}</b>
			<span>({name})</span>
		</div>
	);
}

export default Profie;
