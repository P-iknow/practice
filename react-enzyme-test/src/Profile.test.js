import React from 'react';
import { mount } from 'enzyme';
import Profile from './Profile';

describe('<Profile />', () => {
	it('matches snapshot', () => {
		const wrapper = mount(<Profile userName="P-iknow" name="이영창" />);
		expect(wrapper).toMatchSnapshot();
	});

	it('it render with userName and name', () => {
		const wrapper = mount(<Profile userName="P-iknow" name="이영창" />);
		expect(wrapper.props().userName).toBe('P-iknow');
		expect(wrapper.props().name).toBe('이영창');

		const boldElement = wrapper.find('b');
		expect(boldElement.contains('P-iknow')).toBe(true);
		const spanElement = wrapper.find('span');
    expect(spanElement.text()).toBe('(이영창)');

	});
});
