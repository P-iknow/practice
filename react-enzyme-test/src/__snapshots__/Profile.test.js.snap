// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Profile /> matches snapshot 1`] = `
<Profie
  name="이영창"
  userName="P-iknow"
>
  <div>
    <b>
      P-iknow
    </b>
    <span>
      (
      이영창
      )
    </span>
  </div>
</Profie>
`;
